rspace/src/main/scala/coop/rchain/rspace/nextgenrspace/RSpace.scala

```diff
@@ -210,10 +197,19 @@ class RSpace[F[_], C, P, A, R, K] private[rspace] (
                                       _ <- metricsF.incrementCounter(consumeCommLabel)
                                       _ <- syncF.delay {
                                             eventLog.update(
-                                              COMM(consumeRef, dataCandidates.map(_.datum.source)) +: _
+                                              COMM(
+                                                consumeRef,
+                                                dataCandidates.map(_.datum.source),
+                                                peeks.toSeq.sorted
```

Comment:
I see now. This looks costly. Better enforce `SortedSet` in the method's signature and use the same in `COMM`

Reply:
on it